### ISSUE TITLE
Plex autoscan connection refused

### ISSUE THREAD

**Issue by shanehughes1990** [HaveAGitGat/Tdarr_Plugins#124]
Its likely an issue on my part but for whatever reason I cannot seem to get the plex_autoscan function to work.
The following is a picture of my transcoding steps, plex_autoscan being the last step. All of my media center is running in docker using docker-compose, I have specified a static IP for the plex container so it will _ALWAYS_ come up with that ip-address. I got my token from showing XML of a video on my server and that token works else where, 

This is the output from the debug log on tdarr server
```
request next
[Function (anonymous)]
statusCode: 401
<html><head><title>Unauthorized</title></head><body><h1>401 Unauthorized</h1></body></html>
```

**Comment by shanehughes1990** [comment]
This is hilarious, I just had an epiphany, this is to point to plex_autoscan service _NOT_ plex itself correct? I need another docker container hosting the autoscan service? and this plugin pointing to that?

**Comment by bencmeyer** [comment]
Did you ever get this to work?
I have been messing with this for a day or two and can't figure it out.

I made a reddit post about it;
https://www.reddit.com/r/Tdarr/comments/motfm2/migz_plex_autoscan/

**Comment by karan** [comment]
I tried to make https://gist.github.com/karan/86533635c26c0f8a1b2a7c59c41c12b4 work but my autoscan server doesn't seem to get the directory:

```
Sep 26 13:28:25 INF Scan moved to processor method=POST path=undefined request_id=c58dfqfvmd3f97uu6460 url=/triggers/manual
```

I can however manually and successfully trigger autoscan using `curl --request POST   --url 'http://localhost:3030/triggers/manual?dir=$DIRECTORY`. So I'm at a loss for why the plugin doesn't work.

@Migz93 any ideas?

**Comment by Migz93** [comment]
For the original issue from shanehughes1990 in this thread - correct, my plugin needs the plex autoscan script setup to work, it isn't pointed at plex directly.
For the issue from bencmeyer - my plugin is for plex autoscan, bencmeyer was trying to use it on autoscan which is the newer replacement of plex autoscan.

Just thought i'd mention the above two even though they were a while back incase HaveAGitGat ever wants to close this issue as those two weren't actually problems with the script.


For you @karan, unfortunately I don't use plex autoscan or autoscan anymore and my javascript was never that great so I just pieced together my plugins based on other plugins so I wouldn't really have an idea.
Only thing I do notice though is that in my plugin I'm using the following to set the variable for the file location
```
let filepath = '';
filepath = `${file.file}`;
```

but then in yours you're using
```
const DIRECTORY = file.meta.Directory;
```
I'm guessing maybe file.meta.Directory doesn't have the filepath in and that's why it's coming up as undefined?
Maybe try the same method I used, if the filepath goes through in the correct format for autoscan?

Otherwise beyond that I really have no idea, I remember when I was creating my plugins and ran into problems like this I'd find a way to "echo" the variable so I could see what it's actually set to before I used it anywhere. I can't remember exactly what commands I used but I probably would've found them with a google and then just watched tdarr logs to see it.
There might have been better ways to debug plugin creation but like I said my javascript knowledge was almost non existent.
I hope something I said above helps but sorry I can't help more, I truly have no idea.

**Comment by karan** [comment]
Yeah that's the first thing I tried (your way). I'm not good with js either and can't really get any `console.log` out in tdarr. I'll search around and see if I can find anything helpful. Thanks though.
